src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs

```diff
@@ -916,6 +916,62 @@ public async Task DATA_Received_StreamClosed_ConnectionError()
                 });
         }
 
+        [Fact]
+        public async Task Frame_MultipleStreams_CanBeCreatedIfClientCountIsLessThanActualMaxStreamCount()
+        {
+            _serviceContext.ServerOptions.Limits.Http2.MaxStreamsPerConnection = 1; 
+            var tcs = new TaskCompletionSource<object>();
+            var tcs2 = new TaskCompletionSource<object>();
+            var makeFirstRequestWait = false;
+            await InitializeConnectionAsync(async context =>
+            {
+                if (!makeFirstRequestWait)
+                {
+                    makeFirstRequestWait = true;
+                    tcs2.SetResult(null);
+                    await tcs.Task;
+                }
+            });
+
+            await StartStreamAsync(1, _postRequestHeaders, endStream: true);
+            await SendRstStreamAsync(1);
+
+            await tcs2.Task;
+
+            await StartStreamAsync(3, _postRequestHeaders, endStream: true);
```

Comment:
Why are you using Post with endStream?

Reply:
I just picked a default set of headers. Is there one that is equivalent to "empty"??

Reply:
Seems like browser is the default

Reply:
Yeah _browserRequestHeaders is a GET